source/adios2/CMakeLists.txt

```diff
@@ -120,10 +121,38 @@ if(ADIOS2_HAVE_CUDA)
     OUTPUT_NAME adios2${ADIOS2_LIBRARY_SUFFIX}_core_cuda
     )
   target_compile_features(adios2_core_cuda PRIVATE cuda_std_14)
-  target_link_libraries(adios2_core PRIVATE adios2_core_cuda CUDA::cudart)
+  target_link_libraries(adios2_core PUBLIC adios2_core_cuda PRIVATE CUDA::cudart)
   set(maybe_adios2_core_cuda adios2_core_cuda)
 endif()
 
+set(maybe_adios2_core_kokkos)
+if(ADIOS2_HAVE_Kokkos)
+  add_library(adios2_core_kokkos helper/adiosKokkos.h helper/adiosKokkos.cpp)
+
+  set_target_properties(adios2_core_kokkos PROPERTIES
+    VISIBILITY_INLINES_HIDDEN ON
+    INCLUDE_DIRECTORIES "$<BUILD_INTERFACE:${ADIOS2_SOURCE_DIR}/source>;$<BUILD_INTERFACE:${ADIOS2_BINARY_DIR}/source>"
+    EXPORT_NAME core_kokkos
+    OUTPUT_NAME adios2${ADIOS2_LIBRARY_SUFFIX}_core_kokkos
+    )
+
+  if(Kokkos_ENABLE_CUDA)
+    set_property(SOURCE helper/adiosKokkos.cpp PROPERTY LANGUAGE CUDA)
+    set_property(SOURCE helper/adiosKokkos.cpp APPEND PROPERTY COMPILE_FLAGS "--extended-lambda")
+
+    set_target_properties(adios2_core_kokkos PROPERTIES
+      CUDA_SEPARABLE_COMPILATION ON
+      CUDA_VISIBILITY_PRESET hidden
```

Comment:
Is there a reason why we are setting this to hidden? 
Also why is the separate compilation set to ON?